### PR TITLE
GM: Resolve preenable cruise fault

### LINKED ISSUE #28680: GM sends nonzero gas while preEnabled
### Describe the bug

GM's carcontroller [sends nonzero gas in the preEnable state](https://github.com/commaai/openpilot/blob/master/selfdrive/car/gm/carcontroller.py#L88), i.e. gas is commanded despite the user's foot being on the brake pedal. This is due to only checking if `CC.longActive == True`, even though this may be the case while `CS.brakePressed == True` (in other words, while pre-enabled).

I haven't looked closely into if/how OP handles the preEnable state for other makes, but I believe that this has not been reported earlier because many GM vehicles have their minEnableSpeed set unnecessarily high, despite most being capable of enabling from zero.

This contributes to, though is not the sole cause of, the cruise fault we see when enabling from standstill-- the ECM faults if gas or brake commands are sent while user brake is active. 

Allowing nonzero gas while user brake is pressed appears to be the root cause on the openpilot side of [the same safety issue](https://github.com/commaai/panda/pull/1230) that I observed a few months ago.

### Which car does this affect?

Chevrolet Volt 2018

### Provide a route where the issue occurs

806907cd33ef2647|2023-06-24--09-43-33

### openpilot version

0.9.2

### Additional info

This bug affects the Acadia, Escalade and Escalade ESV in release. The enable speed is set unnecessarily high on most other GM vehicles, including my own, so I had to lower it in order to reproduce the issue. (I suppose I could have achieved the same result by manually fingerprinting as one of those.) I also needed to remove the dashcam flag, since the `dashcamOverride` param was removed in https://github.com/commaai/openpilot/pull/27509. This is why git is dirty in the route I uploaded.

### PR DESCRIPTION
**Pull request by nworb-cire**
Proof of concept currently. Allows pre-enable and removal of user brake into autohold brake mode, but faults once the gas command is sent. Still need to figure that out. 

Should resolve https://github.com/commaai/openpilot/issues/28680 as well, since that seems to be one of the many causes of this fault.

### CODE REVIEW

File: selfdrive/car/gm/gmcan.py [hunk 1]
```diff
@@ -81,7 +81,9 @@ def create_friction_brake_command(packer, bus, apply_brake, idx, enabled, near_s
 
   if apply_brake > 0:
     mode = 0xa
-    if at_full_stop:
+    if pre_enabled:
+      mode = 0xb
+    elif at_full_stop:
       mode = 0xd
```
Comment (sshane) [inline selfdrive/car/gm/gmcan.py:87]: Can you try to send brake without making changes to the mode?

Comment (nworb-cire) [reply selfdrive/car/gm/gmcan.py:87]: I don't have an ASCM/stock ACC, but I could probably ask one of the other guys to record a route.

0xb by itself will allow pre-enable, then holds brakes for roughly half a second after user brake is released. After that, the system faults and releases the brake hold.

0xd by itself causes a fault while user brake is pressed IIRC.

Comment (sshane) [reply selfdrive/car/gm/gmcan.py:87]: I'm surprised this isn't more widely reported then, are you sure it's not due to your car being non-stock? @Verylukyguy can you enable at a stop with your foot on the brake? What happens when you do so?

Comment (Verylukyguy) [reply selfdrive/car/gm/gmcan.py:87]: > 

I can enable at a stop with my foot on the brake.  I can provide a test route, but not until later tonight.

**Comment by sshane** [comment]
Not being able to auto-resume shouldn't be related to this fault, is it?

**Comment by nworb-cire** [comment]
I suppose they are similar issues-- both in preenable and at ACC standstill while already enabled, we would like the car to respond to ACC gas commands, but it either faults or does not respond.

They are not the same because there appear to be three subtly different cruise faults:
1. a cruise fault occurs when nonzero gas is sent while user brake is pressed (https://github.com/commaai/openpilot/issues/28680)
2. a cruise fault occurs when enabling/re-enabling at a perfect standstill (that's what this PR aims to address). No cruise fault occurs if vEgo >= some epsilon > 0
3. a cruise fault occurs when ACC is already enabled, at ACC standstill, and we send a gas command. Some preliminary work has been done to spam resume, but with no results as of yet. (We know we can get the ECU to respond to resume spamming in some cases, see OPGM CC long, aka redneck ACC)

Cases 2 and 3 are slightly different faults because of the vehicle's internal ACC state. In case 3, we can press resume to begin sending gas commands. In case 2, we cannot.

**Comment by nworb-cire** [comment]
Closing pending further investigation.

**Pull request closed without merging**